omnigibson/object_states/slicer_active.py

```diff
@@ -0,0 +1,85 @@
+import numpy as np
+from omnigibson.macros import create_module_macros
+from omnigibson.object_states.object_state_base import AbsoluteObjectState
+from omnigibson.object_states.aabb import AABB
+from omnigibson.object_states.contact_bodies import ContactBodies
+from omnigibson.object_states.update_state_mixin import UpdateStateMixin
+import omnigibson as og
+
+
+# Create settings for this module
+m = create_module_macros(module_path=__file__)
+
+class SlicerActive(AbsoluteObjectState, UpdateStateMixin):
+    @classmethod
+    def get_dependencies(cls):
+        deps = super().get_dependencies()
+        deps.add(AABB)
+        deps.add(ContactBodies)
+        return deps
+
+    def __init__(self, obj):
+        super(SlicerActive, self).__init__(obj)
+
+        self._previously_touching_sliceable = False
+        self.value = True
+        self.delay_counter = 0
+        self.delay_steps = 7
+
+    def _get_value(self):
+        return self.value
+
+    def _set_value(self, new_value):
+        self.value = new_value
+        return True
+    
+    def _currently_touching_sliceable(self):
+        contact_links = self.obj.states[ContactBodies].get_value()
+        contact_link_prim_paths = {contact_link.prim_path for contact_link in contact_links}
+        for prim_path in contact_link_prim_paths:
+            obj_prim_path, _ = prim_path.rsplit("/", 1)
+            candidate_obj = og.sim.scene.object_registry("prim_path", obj_prim_path, None)
+            if candidate_obj is None:
+                continue
+            if hasattr(candidate_obj, '_abilities') and "sliceable" in candidate_obj._abilities:
+                return True
+        return False
+
+    def _update(self):
+        # If we were slicing in the past step, deactivate now.
+        if self._previously_touching_sliceable:
+            self.set_value(False)
+            self.delay_counter = 0  # Reset the counter when we stop touching a sliceable object
+
+        # Are we currently touching any sliceables?
+        currently_touching_sliceable = self._currently_touching_sliceable()
+
+        # If our value is False, we need to consider reverting back.
+        if not self.value:
+            # If we are not touching any sliceable objects, we can revert to True.
+            if not currently_touching_sliceable:
```

Comment:
The `else` case of this happens in the next step's line 52, but maybe let's still add it here - if we are touching sliceable, just reset the timer.